### PR TITLE
fix: clean up rebass

### LINKED ISSUE #294: rebass is installed
### Subject of the issue

One thing I noticed is that `rebass` is an [`optionalDependencies `](https://github.com/mdx-js/eslint-mdx/blob/master/packages/eslint-plugin-mdx/package.json#L46), so yarn is going to install it whenever possible as per https://classic.yarnpkg.com/en/docs/dependency-types/, and it installs quite a lot of packages https://github.com/webpack/webpack.js.org/pull/4326/files#diff-51e4f558fae534656963876761c95b83b6ef5da5103c4adef6768219ed76c2de.

But I don't use `rebass` at all, hence it just doesn't make sense.

### Steps to reproduce

Just install `eslint-plugin-mdx` and check `yarn.lock` for `rebass`.

### Expected behaviour

`rebass` should not be installed just because I have `eslint-plugin-mdx` installed.

### Actual behaviour

`rebass` is installed along with `eslint-plugin-mdx`.

### PR DESCRIPTION
**Pull request by chenxsan**
Closes [GH-294](https://github.com/mdx-js/eslint-mdx/issues/294).

### CODE REVIEW

**Comment by JounQin** [comment]
We don’t put dev in workspace packages, move it to root package.json instead.

**Comment by JounQin** [comment]
I thought `rebass` been bundled with `mdx` itself because it is mentioned in the playground, if it’s not, then `rebass` related codes should all be remove in this repo.

@ChristianMurphy Is that true?

**Comment by chenxsan** [comment]
Moving to the root `package.json` will cause eslint to show errors:

image

Does it mean I have to update eslint configuration as well?

**Comment by JounQin** [comment]
As described at https://github.com/mdx-js/eslint-mdx/pull/295#issuecomment-803251674

You can remove all codes related to `rebass`, and remove `rebass` as dep, not even dev.

https://github.com/mdx-js/eslint-mdx/blob/master/packages/eslint-plugin-mdx/src/configs/overrides.ts

**Comment by JounQin** [comment]
`@types/rebass` should be removed too.

https://github.com/mdx-js/eslint-mdx/blob/master/package.json#L33

File: packages/eslint-plugin-mdx/src/configs/overrides.ts [hunk 1]
```diff
@@ -1,25 +1,9 @@
...
-  rebass = require('rebass') as typeof rebass
-} catch {
-  // `rebass`(or `reflexbox` actually) requires `react` as peerDependency, but not all projects using `mdx` are `React` based, so we fallback to hardcoded `rebass` Components here
-  /* istanbul ignore next */
-  rebass = ['Box', 'Flex', 'Text', 'Heading', 'Link', 'Button', 'Image', 'Card']
-}
 
 export const overrides: Linter.Config = {
   ...base,
-  globals: getGlobals(rebass, {
```
Comment (JounQin) [inline packages/eslint-plugin-mdx/src/configs/overrides.ts:20]: `React` should be available globally.

**Comment by JounQin** [comment]
Thanks for cleaning up!

**Pull request merged**